tests/test_shipment.py

```diff
@@ -1,228 +1,157 @@
+    so that USPS doesn't automatically insure it so we could manually insure it here.
+    """
+    shipment_data = one_call_buy_shipment
+    # Set to 0 so USPS doesn't insure this automatically and we can insure the shipment manually
+    shipment_data["insurance"] = 0
+
+    shipment = easypost.Shipment.create(**shipment_data)
+    shipment.insure(amount="100")
+
+    assert shipment.insurance == "100.00"
 
-    shipment.regenerate_rates()
 
-    new_rate_id = shipment.rates[0].id
-    assert new_rate_id is not None
-    assert new_rate_id != rate_id
+@pytest.mark.vcr()
+def test_shipment_refund(one_call_buy_shipment):
+    """
+    Refunding a test shipment must happen within seconds of the shipment being created as test shipments naturally
+    follow a flow of created -> delivered to cycle through tracking events in test mode - as such anything older
+    than a few seconds in test mode may not be refundable.
+    """
+    shipment = easypost.Shipment.create(**one_call_buy_shipment)
+    shipment.refund()
+
+    assert shipment.refund_status == "submitted"
 
 
 @pytest.mark.vcr()
-def test_smartrate(vcr):
-    to_address = {
-        "name": "Dr. Steve Brule",
-        "street1": "179 N Harbor Dr",
-        "city": "Redondo Beach",
-        "state": "CA",
-        "zip": "90277",
-        "country": "US",
-        "phone": "4153334444",
-        "email": "dr_steve_brule@gmail.com",
-    }
-    from_address = {
-        "name": "EasyPost",
-        "street1": "417 Montgomery Street",
-        "street2": "5th Floor",
-        "city": "San Francisco",
-        "state": "CA",
-        "zip": "94104",
-        "country": "US",
-        "phone": "4153334444",
-        "email": "support@easypost.com",
-    }
-    parcel = {
-        "length": 20.2,
-        "width": 10.9,
-        "height": 5,
-        "weight": 65.9,
-    }
+def test_shipment_smartrate(basic_shipment):
+    shipment = easypost.Shipment.create(**basic_shipment)
 
-    shipment = easypost.Shipment.create(
-        to_address=to_address,
-        from_address=from_address,
-        parcel=parcel,
-    )
-    assert shipment.rates
+    assert shipment.rates is not None
 
     smartrates = shipment.get_smartrates()
-    assert shipment.rates[0]["id"] == smartrates[0]["id"]
+
     assert smartrates[0]["time_in_transit"]["percentile_50"] is not None
     assert smartrates[0]["time_in_transit"]["percentile_75"] is not None
     assert smartrates[0]["time_in_transit"]["percentile_85"] is not None
     assert smartrates[0]["time_in_transit"]["percentile_90"] is not None
     assert smartrates[0]["time_in_transit"]["percentile_95"] is not None
     assert smartrates[0]["time_in_transit"]["percentile_97"] is not None
     assert smartrates[0]["time_in_transit"]["percentile_99"] is not None
+
+
+@pytest.mark.vcr()
+def test_shipment_create_empty_objects(basic_shipment):
+    shipment_data = basic_shipment
```

Comment:
You also need to create this shipment with an empty list for `customs_info` to test we don't break things with empty lists. You'll then need to assert that.

Reply:
The assertions are correct and passed, I didn't see why we need it since it's already null. Anyway, I will add this back.

Reply:
I understand these assertions are correct. I'm asking that **in addition** to these assertions you already have, you should also pass an empty list to `customs_info`. This tests purpose is to check that we encode data to the API correctly. We had big problems with the PHP encoding because we didn't test enough edge cases (users frequently pass us empty lists) and so we should test/guard against that by having a test that does this as we do in all our other libs.

Reply:
Thanks for the explanation, I have added an empty list to the customs_info field.